spec/support/shared_examples/ca.rb

```diff
@@ -0,0 +1,88 @@
+shared_examples "a CA" do
+
+  it { is_expected.to respond_to(:create) }
+  it { is_expected.to respond_to(:create_keypair) }
+  it { is_expected.to respond_to(:create_keypair_in) }
+  it { is_expected.to respond_to(:reference_keypair) }
+  it { is_expected.to respond_to(:<=>) }
+
+  before do
+    @ca = described_class.create("some-ca", "a-bucket")
+  end
+
+  describe ".create" do
+
+    it "should put the cert in s3" do
+      ca_cert = @ca.output["resource"]["aws_s3_bucket_object"].select { |_, obj|
+        obj[:key] == "/some-ca/ca.cert" && obj[:bucket] == "a-bucket"
+      }
+
+      expect(ca_cert.count).to eq(1)
+      expect(@ca.source).to eq("s3://a-bucket/some-ca/ca.cert")
+    end
+
+    it "should populate name" do
+      expect(@ca.name).to eq("some-ca")
+    end
+
+  end
+
+  describe ".create_keypair_in" do
+
+    it "should put stuff in the right context" do
+      ctx = Terrafying::Context.new
+
+      keypair = @ca.create_keypair_in(ctx, "foo")
+
+      resource_names = keypair[:resources].map { |r| r.split(".")[1] }
+
+      expect(ctx.output["resource"]["aws_s3_bucket_object"].keys).to include(*resource_names)
+      expect(@ca.output["resource"]["aws_s3_bucket_object"].keys).to_not include(*resource_names)
+    end
+
+  end
+
+  describe ".create_keypair" do
+
+    it "should reference the right bucket objects in output" do
+      keypair = @ca.create_keypair("foo")
+
+      cert_object = @ca.output["resource"]["aws_s3_bucket_object"].select { |_, obj|
+        File.join("s3://", obj[:bucket], obj[:key]) == keypair[:source][:cert]
+      }
+      key_object = @ca.output["resource"]["aws_s3_bucket_object"].select { |_, obj|
+        File.join("s3://", obj[:bucket], obj[:key]) == keypair[:source][:key]
+      }
+
+      expect(cert_object).to_not be nil
```

Comment:
Is this sufficient to know that it "reference[s] the right bucket objects in output"? Can we test the actual object?